src/solvers/petscdmlibmeshimpl.C

```diff
@@ -512,12 +510,12 @@ PetscErrorCode DMlibMeshCreateFieldDecompositionDM(DM dm, PetscInt dnumber, Pets
       }
     }
   }
-  else { /* Empty splits indicate default: split all variables with one per split. */
+  else { // Empty splits indicate default: split all variables with one per split.
     PetscInt d = 0;
-    for (std::map<std::string, unsigned int>::const_iterator vit = ddlm->varids->begin(); vit != ddlm->varids->end(); ++vit) {
+    for (const auto & pr : (*ddlm->varids)) {
       ddlm->decomposition->push_back(std::set<unsigned int>());
-      unsigned int vid = vit->second;
-      std::string vname = vit->first;
+      unsigned int vid = pr.second;
+      std::string vname = pr.first;
```

Comment:
How is this not throwing up an unused variable warning?

Reply:
No clue, I thought maybe they were i a section of code that was ifdef'd out but that does not appear to be the case. I'll drop the unused vars...